domain-models-runtime/src/main/resources/templates/db_scripts/audit.ftl

```diff
@@ -10,29 +10,28 @@ CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.audit_${table.tableName}_
   <#if (table.auditingSnippet.insert.declare)??>
     ${table.auditingSnippet.insert.declare}
   </#if>
-    seed TEXT;
-    maxid UUID;
     jsonb JSONB;
+    uuidtext TEXT;
+    uuid UUID;
   BEGIN
-    maxid = (SELECT ${myuniversity}_${mymodule}.max(id) FROM ${myuniversity}_${mymodule}.${table.auditingTableName});
-    IF maxid IS NULL THEN
-      -- don't use random, that will be different on each pg-pool replication node
-      seed = md5(concat('${myuniversity}_${mymodule}.${table.auditingTableName}', NEW.jsonb));
-      -- avoid overflow
-      IF (left(seed, 13) = 'ffffffff-ffff') THEN
-        seed = overlay(seed placing '00000000-0000' from 1);
-      END IF;
-      -- UUID version byte
-      seed = overlay(seed placing '4' from 13);
-      -- UUID variant byte
-      seed = overlay(seed placing '8' from 17);
-      maxid = seed::uuid;
-    ELSE
-      maxid = ${myuniversity}_${mymodule}.next_uuid(maxid);
-    END IF;
+    jsonb = CASE WHEN TG_OP = 'DELETE' THEN OLD.jsonb ELSE NEW.jsonb END;
+
+    -- create uuid based on the jsonb value so that concurrent updates of different records are possible.
+    uuidtext = md5(jsonb::text);
```

Comment:
Any performance concern to run md5 for every record?

Reply:
create table t (a int);

    insert into t values (generate_series(1, 1000000));
    INSERT 0 1000000
    Time: 1157.428 ms (00:01.157)

    insert into t values (length(md5('' || generate_series(1, 1000000))));
    INSERT 0 1000000
    Time: 1667.770 ms (00:01.668)

About half a nanosecond. This is fast enough.